Src/Couchbase.Linq/SaveOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Couchbase.Linq
+{
+    /// <summary>
+    /// Provides options controlling how changes are saved via <see cref="IBucketContext"/>.
+    /// </summary>
+    public class SaveOptions
```

Comment:
@jeffrymorris 

You make a good point.  Part of the problem is that we don't really have a good way to return such a status back to the user, we're taking the more LINQ like approach of throwing exceptions.

The best option I can think of off the top of my head would be:
1.  Make the setting an enumeration with three possible behaviors (ignore, fail and continue, fail and stop)
2.  if fail and stop, throw an immediate exception
3.  if fail and continue, collect the exception, and throw an AggregateException at the end with all collected errors

Thoughts?

Brant